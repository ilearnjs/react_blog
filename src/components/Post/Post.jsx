import React, { Component } from "react";
import { Link } from 'react-router-dom'

class Post extends Component {
	onRemoveClicked() {
		const { post: { id }, remove } = this.props;
		
		remove(id);
	}

	render() {
		const { post, userName } = this.props;

		return (
			<div className="post">
				<div className="post-content">
					<div className="header">
						<div className={`avatar ${post.user.name}`}>
						</div>
						<Link to={`/user/${post.user.name}`}>
							<span className="name">
								{`mr.${post.user.name}`}
							</span>
						</Link>
						<span className="created">
							{post.created.toLocaleDateString()}
						</span>
						{userName === post.user.name &&
							<button className="remove-btn" onClick={() => this.onRemoveClicked()}>
								Remove
							</button>
						}
					</div>
					<div className="text">
						{post.content}
					</div>
				</div>
			</div>
		);
	}
}

export default Post;