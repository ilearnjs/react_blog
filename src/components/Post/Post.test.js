import React from 'react';
import { MemoryRouter } from 'react-router';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';

import Post from './Post';

const post = {
	_id: 332,
	user: {
		name: 'Alisa',
	},
	createdOn: new Date('01-01-2018'),
	content: 'Azaza'
};

const creator = {
	name: 'Alisa',
};

const notCreator = {
	name: 'Bob',
};

describe('Post.jsx', () => {
	test('renders correctly for creator', () => {
		const wrapper = shallow(
			<Post post={post} currentUser={creator} />
		);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	test('renders correctly for not creator', () => {
		const wrapper = shallow(
			<Post post={post} currentUser={notCreator} />
		);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});

	test('remove post called', () => {
		const removePost = jest.fn();

		const wrapper = shallow(
			<Post post={post} currentUser={creator} removePost={removePost} />
		);

		wrapper.find(".remove-btn").simulate("click");
		expect(removePost.mock.calls.length).toBe(1);
	});

	test('remove post called with correct id', () => {
		const removePost = jest.fn();

		const wrapper = shallow(
			<Post post={post} currentUser={creator} removePost={removePost} />
		);

		wrapper.find(".remove-btn").simulate("click");
		expect(removePost.mock.calls[0][0]).toBe(post._id);
	});
});
