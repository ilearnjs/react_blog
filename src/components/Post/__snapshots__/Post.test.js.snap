// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post.jsx renders correctly for creator 1`] = `
<div
  className="post"
>
  <div
    className="post-content"
  >
    <div
      className="header"
    >
      <div
        className="avatar"
        style={
          Object {
            "backgroundImage": "url(https://api.adorable.io/avatars/60/Alisa)",
          }
        }
      />
      <Link
        replace={false}
        to="/user/Alisa"
      >
        <span
          className="name"
        >
          Alisa
        </span>
      </Link>
      <span
        className="created"
      >
        1/1/2018
      </span>
      <button
        className="remove-btn"
        onClick={[Function]}
      >
        Remove
      </button>
    </div>
    <div
      className="text"
    >
      Azaza
    </div>
  </div>
</div>
`;

exports[`Post.jsx renders correctly for not creator 1`] = `
<div
  className="post"
>
  <div
    className="post-content"
  >
    <div
      className="header"
    >
      <div
        className="avatar"
        style={
          Object {
            "backgroundImage": "url(https://api.adorable.io/avatars/60/Alisa)",
          }
        }
      />
      <Link
        replace={false}
        to="/user/Alisa"
      >
        <span
          className="name"
        >
          Alisa
        </span>
      </Link>
      <span
        className="created"
      >
        1/1/2018
      </span>
    </div>
    <div
      className="text"
    >
      Azaza
    </div>
  </div>
</div>
`;
